src/GitHub.TeamFoundation.14/Services/VSGitExt.cs

```diff
@@ -49,29 +53,28 @@ public VSGitExt(IAsyncServiceProvider asyncServiceProvider, IVSUIContextFactory
             // The IGitExt service isn't available when a TFS based solution is opened directly.
             // It will become available when moving to a Git based solution (and cause a UIContext event to fire).
             var context = factory.GetUIContext(new Guid(Guids.GitSccProviderId));
-            context.WhenActivated(() => Initialize());
+            context.WhenActivated(() => JoinableTaskFactory.RunAsync(InitializeAsync));
```

Comment:
I'm not clear on why this is different from just calling `Initialize` as we were previously? The result of `RunAsync` isn't awaited,.

Reply:
It automatically adds running tasks to `JoinableTaskCollection`, which can be awaited by unit tests. It also allows use to `SwitchToMainThreadAsync` in the `GetServiceAsync<T>()` method below.

It we use a common `JoinableTaskContext` across the extension, I'm hoping it will let us more easily debug our task related code and avoid possible deadlocks.

Reply:
asdf

Reply:
test

Reply:
Hello